app/services/robad/resource_management/port.rb

```diff
@@ -3,9 +3,38 @@
 module Robad
   module ResourceManagement
     class Port
+      PORT_RANGE = (10_000..20_000).freeze
+      RESOURCE_ALLOCATED_PORTS_KEY = :resource_allocated_ports
+
       class << self
         def allocate
-          rand(10_000..20_000)
+          new_port = nil
+          RedisStorage.lock!(RESOURCE_ALLOCATED_PORTS_KEY) do
+            current_ports = allocated_ports
+            new_port = (PORT_RANGE.to_a - current_ports).sample
+            RedisStorage[RESOURCE_ALLOCATED_PORTS_KEY] = current_ports.push(new_port)
```

Comment:
It's better to store more extra info like creation time, application and process name, besides the port itself

Reply:
I'll create _cache_constants.rb_ and move RESOURCE_ALLOCATED_PORTS_KEY there
PORT_RANGE may be in port.rb

Reply:
Done